src/main/java/graphql/introspection/Introspection.java

```diff
@@ -702,6 +702,43 @@ public static boolean isIntrospectionTypes(GraphQLNamedType type) {
      */
     public static GraphQLFieldDefinition getFieldDef(GraphQLSchema schema, GraphQLCompositeType parentType, String fieldName) {
 
+        GraphQLFieldDefinition fieldDefinition = getSystemFieldDef(schema, parentType, fieldName);
+        if (fieldDefinition != null) {
+            return fieldDefinition;
+        }
+
+        assertTrue(parentType instanceof GraphQLFieldsContainer, () -> String.format("should not happen : parent type must be an object or interface %s", parentType));
+        GraphQLFieldsContainer fieldsContainer = (GraphQLFieldsContainer) parentType;
+        fieldDefinition = schema.getCodeRegistry().getFieldVisibility().getFieldDefinition(fieldsContainer, fieldName);
+        assertTrue(fieldDefinition != null, () -> String.format("Unknown field '%s' for type %s", fieldName, fieldsContainer.getName()));
+        return fieldDefinition;
+    }
```

Comment:
I have left the old method in place for semantic reasons and also its signature is not what the execution engine needs.  

The `parentType instanceof GraphQLFieldsContainer` is not needed say.